tests/python_package_test/test_basic.py

```diff
@@ -411,3 +411,18 @@ def test_list_to_1d_numpy(y, dtype):
     result = lgb.basic.list_to_1d_numpy(y, dtype=dtype)
     assert result.size == 10
     assert result.dtype == dtype
+
+
+def test_dump_model():
+    X, y = load_breast_cancer(return_X_y=True)
+    train_data = lgb.Dataset(X, label=y)
+    params = {
+        "objective": "binary",
+        "verbose": -1
+    }
+    bst = lgb.train(params, train_data, num_boost_round=5)
+    bst.dump_model(5, 0)
+    params['linear_tree'] = True
+    train_data = lgb.Dataset(X, label=y)
+    bst = lgb.train(params, train_data, num_boost_round=5)
+    bst.dump_model(5, 0)
```

Comment:
Shall we use `loads` to validate that the output JSON content?

Reply:
This is done already in `dump_model`: https://github.com/microsoft/LightGBM/blob/36957ed5fd974680cd2953e3f483e33f36ceb185/python-package/lightgbm/basic.py#L3090